content/blog/setting-up-github-actions-for-v.md

```diff
@@ -0,0 +1,355 @@
+jobs:
+  build:
+    strategy:
+      matrix:
+        os: ["ubuntu-latest", "macos-latest", "windows-latest"]
+    runs-on: ${{ matrix.os }}
+```
+
+Note that _macos_ and _windows_ runners have lower availability than
+_ubuntu_ runners. So it might be worthwhile to quickly prototype on
+`ubuntu-latest` and then add `macos-latest` and/or `windows-latest` when
+needed.
+
+We can now defined the _steps_ for the `build` job:
+
+```yaml {tabWidth=2}
+# ...
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        os: ["ubuntu-latest", "macos-latest", "windows-latest"]
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - name: Install V
+        uses: vlang/setup-v@v1
+        with:
+          check-latest: true
+
+      - name: Checkout ${{ github.event.repository.name }}
+        uses: actions/checkout@v2
+
+      - name: Check if code is formatted
+        run: |
+          v fmt -diff .
+          v fmt -verify .
+
+      - name: Build ${{ github.event.repository.name }}
+        run: v .
+
+      - name: Run Tests
+        run: v test .
+```
+
+First, we use the `vlang/setup-v@1` GitHub Action to install V for our runner.
+We set `check-latest` to `true` to get the latest updates. Leave this out if
+you prefer more stability. Then we clone the current project using the
+`actions/checkout@v2` action.
+
+We can now start checking the codebase to ensure that it passes the minimum
+standards we set. V ships with its own opinionated tool for formatting V source code.
+Here is a sample usage:
+
+```bash
+# Format all V files recursively in this directory
+# AND rewrite them
+v fmt -w .
+# ^^^
+# You'll most likely use this all the time from the root of
+# your V projects.
+
+# Display the formatted output of a sample V file to stdout
+v fmt sample.v
+```
+
+You can run `v help fmt` to get more details on all the available options.
+Two more options we have at our disposal are:
+
+1. `v fmt -diff .` which displays the difference between the original and
+   formatted source, if any. Ideally, we don't any difference before we
+   push the commits.
+2. `v fmt -verify .` which reports an error if it finds an unformatted
+   file in the repository.
+
+Note that we do not automatically format the code in the CI because it might
+lead to huge `diff`s later on and a lot of potentially redundant commits. It
+is also easier to have our tests be _read-only_.
+
+The source code formatting check finishes quickly. Next we make sure that we
+can build our project. This is as simple as running the `v .` command. This
+will build the project fast. If you plan to distribute the built executable,
+it is recommended to use `v -prod .` to generate an optimised executable,
+which takes slightly longer. You can also choose to generate optimised `-prod`
+builds, only when a tag is pushed, instead of on each commit or Pull Request.
+
+Finally, we run the unit tests (if any) to make sure that we have not
+regressed, in the form of failing tests. V's in-built testing framework
```

Comment:
```suggestion
Finally, we run the unit tests (if any) to make sure that we have not
regressed. This will be shown by failing tests. V's built-in testing framework
```
FYI... `in-built` is often used to mean the same as `built-in`, but `built-in` is much more popular usage (`in-built` is mostly only use in the UK).  Some consider them to be slightly different meaning... `in-build` meaning more of an innate piece of the original (like your heart and lungs), while `built-in` is extra parts added to the original (cyborgs?  :-)).